src/main/scala/nl/knaw/dans/easy/stage/EasyStageDataset.scala

```diff
@@ -116,6 +116,7 @@ object EasyStageDataset {
         bagRelativePath = s.bagitDir.toPath.relativize(file.toPath).toString
         mime <- readMimeType(bagRelativePath)
         title <- readTitle(bagRelativePath)
+        fileAccessRights <- getFileAccessRights(bagRelativePath)
```

Comment:
You are indirectly calling `readFileMetadata` 3 times. Please reduce to one call.